backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/ElasticsearchLoadableSearchResult.java

```diff
@@ -57,7 +57,8 @@
 	ElasticsearchSearchResult<H> loadBlocking() {
 		SearchProjectionTransformContext transformContext = extractContext.createProjectionTransformContext();
 
-		LoadingResult<?> loadingResult = extractContext.getProjectionHitMapper().loadBlocking();
+		// TODO HSEARCH-3352 pass timeout
+		LoadingResult<?> loadingResult = extractContext.getProjectionHitMapper().loadBlocking( null );
```

Comment:
Same problem as for the multi-entity loading: if you have a timeout of 2ms, and you already spent 1ms retrieving the results from Elasticsearch, you're allowing loading to take 2ms, so in the end you allow 3ms.

To address this, you shouldn't pass a timeout, you should pass a deadline (Instant or long representing the nanos, whatever you prefer). Or if you really want to pass a timeout, you should subtract the elapsed time.